### PR TITLE
#184752250 feature(set Roles):admin should be able to set Roles and permission

### PR DESCRIPTION
**Pull request by RichardIshimwe**

### CODE REVIEW

File: src/database/migrations/20230322155803-create-user.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ module.exports = {
       preferredCurrency: DataTypes.STRING,
       location: DataTypes.STRING,
       billingAddress: DataTypes.STRING,
-      role: {
+      roleId: {
```
Comment (RichardIshimwe) [inline src/database/migrations/20230322155803-create-user.js]: adding roleId as a foreign key

File: lib/database/migrations/20230322155803-create-user.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ module.exports = {
       preferredCurrency: DataTypes.STRING,
       location: DataTypes.STRING,
       billingAddress: DataTypes.STRING,
-      role: {
+      roleId: {
```
Comment (RichardIshimwe) [inline lib/database/migrations/20230322155803-create-user.js]: adding roleId as a foreign key

File: lib/controllers/user/forgotPassword.controller.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ const forgotPassword = async (req, res) => {
       }, resetSecret, {
         expiresIn: '10m'
       });
-      const link = `http://localhost:${process.env.PORT}/api/v1/user/reset-password/${user.id}/${token}`;
+      const link = `http://${process.env.HOST}:${process.env.PORT}/api/v1/user/reset-password/${user.id}/${token}`;
```
Comment (RichardIshimwe) [inline lib/controllers/user/forgotPassword.controller.js:37]: shift the hostname in .env

Comment (RichardIshimwe) [inline lib/controllers/user/forgotPassword.controller.js:37]: shift host in .env

File: lib/database/models/user.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ module.exports = (sequelize, DataTypes) => {
     preferredCurrency: DataTypes.STRING,
     location: DataTypes.STRING,
     billingAddress: DataTypes.STRING,
-    role: {
+    roleId: {
```
Comment (RichardIshimwe) [inline lib/database/models/user.js]: roleId as a foreign key

File: src/database/models/user.js [hunk 5]
```diff
@@ -39,4 +39,4 @@ module.exports = (sequelize, DataTypes) => {
     },
   );
   return User;
-};
+};
```
Comment (RichardIshimwe) [inline src/database/models/user.js]: leaving a space below according to eslint

File: lib/controllers/role.controller.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    if (!user) {
+      return res.status(404).send(`${email} is not found`);
+    }
+    user.roleId = role;
+    await user.save();
+    return res.status(200).send(user);
+  }
+}
+var _default = roleToSet;
+exports.default = _default;
```
Comment (RichardIshimwe) [inline lib/controllers/role.controller.js]: lib file

File: lib/controllers/user/forgotPassword.controller.js [hunk 7]
```diff
@@ -34,7 +34,7 @@ const forgotPassword = async (req, res) => {
       }, resetSecret, {
         expiresIn: '10m'
       });
-      const link = `http://localhost:${process.env.PORT}/api/v1/user/reset-password/${user.id}/${token}`;
```
Comment (RichardIshimwe) [inline lib/controllers/user/forgotPassword.controller.js:37]: diff

File: lib/database/migrations/20230322155803-create-user.js (hunk 2 above)
Comment (RichardIshimwe) [inline lib/database/migrations/20230322155803-create-user.js]: diff

File: lib/database/models/user.js (hunk 4 above)
Comment (RichardIshimwe) [inline lib/database/models/user.js]: diff

File: src/database/models/user.js (hunk 5 above)
Comment (RichardIshimwe) [inline src/database/models/user.js]: diff

Comment (RichardIshimwe) [inline src/database/models/user.js]: diff

File: lib/controllers/user/forgotPassword.controller.js (hunk 7 above)
Comment (RichardIshimwe) [inline lib/controllers/user/forgotPassword.controller.js:37]: diff

File: package.json [hunk 8]
```diff
@@ -52,6 +52,9 @@
     "nodemailer": "^6.9.1",
     "nodemailer-express-handlebars": "^6.1.0",
     "nodemon": "^2.0.21",
+    "passport": "^0.6.0",
+    "passport-google-oauth20": "^2.0.0",
```
Comment (RichardIshimwe) [inline package.json]: diff

File: package.json [hunk 9]
```diff
@@ -52,6 +52,9 @@
...
+    "passport-microsoft": "^1.0.0",
```
Comment (RichardIshimwe) [inline package.json]: diff

**Review by Gentille-dev — APPROVED** [review]
check

**Review by madebypatrick — APPROVED** [review]

**Review by EdwinBayingana — APPROVED** [review]

**Review by Folio123-c — APPROVED** [review]

**Review by Cynthia-kam — APPROVED** [review]

File: lib/database/config/config.js [hunk 10]
```diff
@@ -20,13 +21,14 @@ module.exports = {
     use_env_variable: '',
     username: process.env.DB_USER,
     password: process.env.DB_PASSWORD,
-    database: process.env.TEST_DB_NAME,
+    database: process.env.DB_NAME,
```
Comment (RichardIshimwe) [inline lib/database/config/config.js]: diff

File: lib/database/config/config.js [hunk 11]
```diff
@@ -36,16 +38,18 @@ module.exports = {
     dialect: 'postgres',
     dialectOptions: {
       ssl: {
-        rejectUnauthorized: true
+        require: false,
```
Comment (RichardIshimwe) [inline lib/database/config/config.js]: diff

File: lib/database/config/config.js [hunk 12]
```diff
@@ -36,16 +38,18 @@ module.exports = {
...
+        rejectUnauthorized: false
       }
     }
   },
   staging: {
-    use_env_variable: 'DATABASE_URL_STAGING',
+    use_env_variable: 'DATABASE_URL',
```
Comment (RichardIshimwe) [inline lib/database/config/config.js]: diff

File: lib/database/config/config.js [hunk 13]
```diff
@@ -36,16 +38,18 @@ module.exports = {
...
     dialect: 'postgres',
     dialectOptions: {
       ssl: {
-        rejectUnauthorized: true
+        require: false,
+        rejectUnauthorized: false
```
Comment (RichardIshimwe) [inline lib/database/config/config.js]: diff

File: lib/controllers/user/forgotPassword.controller.js (hunk 7 above)
Comment (RichardIshimwe) [inline lib/controllers/user/forgotPassword.controller.js:37]: marked

**Pull request merged**